src/lib/Config.ts

```diff
@@ -0,0 +1,120 @@
+export default class Config {
+
+    // Signal if the app should be started in offline mode
+    public static offline = navigator.onLine !== undefined && !navigator.onLine;
+
+    // When packaged as distributed offline app, subdomains are folder names instead
+    public static offlinePackaged = false;
+
+    /* Public methods */
+
+    public static origin(subdomain: string) {
+        return Config._origin(subdomain);
+    }
+
+    public static src(subdomain: string) {
+        return Config._origin(subdomain, true);
+    }
+
+    public static get tld() {
+        const tld = Config._myOrigin.split('.');
+        return [tld[tld.length - 2], tld[tld.length - 1]].join('.');
+    }
+
+    public static get network() {
+        if (Config.offlinePackaged) { return 'main'; }
+
+        switch (Config.tld) {
+            case 'nimiq.com': return 'main';
+            case 'nimiq-testnet.com': return 'test';
+            default: return 'test'; // Set this to 'test', 'bounty', or 'dev' for localhost development
+        }
+    }
+
+    public static get cdn() {
+        if (Config.offlinePackaged) { return Config.src('keyguard') + '/nimiq.js'; }
+
+        switch (Config.tld) {
+            case 'nimiq.com': return 'https://cdn.nimiq.com/nimiq.js';
+            default: return 'https://cdn.nimiq-testnet.com/nimiq.js';
+        }
+    }
+
+    public static set devMode(devMode) {
+        Config.isDevMode = devMode;
+    }
+
+    public static get devMode() {
+        if (Config.isDevMode) { return Config.isDevMode; }
+
+        switch (Config.tld) {
+            case 'nimiq.com':
+            case 'nimiq-testnet.com':
+                return false;
+            default:
+                return true;
+        }
+    }
+
+    public static get online() {
+        return !Config.offline;
+    }
+
+    public static get nimiqOrigins() {
+        return [
+            Config.origin('safe'),
+            Config.origin('miner'),
+            Config.origin('promo'),
+            // Config.origin('shop'),
+        ];
+    }
+
+    /* Private methods */
+
+    private static isDevMode: boolean = false;
+
+    private static _origin(subdomain: string, withPath?: boolean) {
+        if (Config._myOrigin.includes('localhost')) {
+            return Config._localhost(subdomain, withPath);
+        }
+
+        if (Config.devMode) {
+            return Config._localhost(subdomain, withPath, true);
+        }
+
+        return `https://${subdomain}.${Config.tld}${(withPath && '/') || ''}`;
```

Comment:
I would use the ternary if there: `withPath ? '/' : ''`. I think it is much easier to understand.